### PR TITLE
[s] Super's Christmas Bash!

### PR DESCRIPTION
**Pull request by Super3222**
#### Intent of your Pull Request

# Christmas at Yogs!

## EGGNOG

- Eggnog bottles spawn in the bartenders vendor.
- Pour eggnog into a shot glass to create a Chrismtas Miracle. Enjoy!

## MAP

*Utahclock's invention. Tweaked and Fixed by Super3222*

The map is full of SNOW! ( https://github.com/yogstation13/yogstation/pull/1412 )
Mysterious snowmen are amock! Fear them for they are walking!
Enjoy the pretty christmas trees!

## EVENTS

~~Snowmen Hivemind Event~~

~~There's a random event that pisses off all of the snowmen on the map.~~
~~They gain their own faction, become 2x faster, and have a more vicious death message.~~

~~They still do 1 melee of damage. It's just to shock the crew.~~
~~There's also a 1% chance that one of the snowmen will gain sentience.~~

### Santa

Their headbowman is abstract, and can't be removed.
They have a new spell to assign someone to the naughty list. Cast the spell, a piece of coal spawns in Santa's hand, activate it, and it'll open up a box to input a name into. Input a name, if it's real than they're added to the naughty list.

### The Grinch

This is virtually an antag running around and ruining Christmas. They can only steal / mess with stuff, they aren't geared to kill. In fact, I've prompted them with a gigantic message telling them not too.

They are actually a "whatchamickcallit", a green-haired pissed off race of aliens that have to listen all day to a salty anrrator roasting the fuck out of them.

And to top it all off, Santa as put every last one of them on the naughty list. So they're even MORE pissed off.

They get a chameleon suit.
Wehn they're attacked, and health lowered beyond 80 than they'll reveal themselves. Instantly unequipping every last thing on them, and exposing themselves for who they truly are.

They get a spell that changes any item (non high risk) into a SORD of it's former self. In roughly 150 seconds it'll revert back.

Activating this special type of SORD in your hand will give your mind bad thoughts and memories depending on your job on the station.

If Santa activates one of the grinch'd items in their hand it will revert back to it's former self instantly.

`else if (user.job == "")
user << "<span class='warning'></span>"`

If you want to help contribute special messages to **specific** jobs.

I had to improvise for an icon. They initially look like abductors under their cham gear, but when they're exposed they lose all of their clothing (this was apart of the original plan), and are dyed with a green color.

### Christmas Krampus

Oh... jolly.

The Christmas Krampus is running amok on the station.

Becareful, it's an unpredictable one! As the legend goes it hunts down naughty children, and throws them into his bucket of eternal evil!

- Santa can summon coal in his hand, and write someones name in it (activate)
- If that person is alive, they're thrown into the naughty list
- The Christmas Krampus's naughty stare is most effective on naughty children.
- The Christmas Krampus can also throw naughty children into his bag as twice as fast.

Krampus is a demon. They do not need to breathe, and they cannot use guns.

Before I start, I want to say that **everything** for Krampus has been tested, and works.

Krampus spawns with this:
- A santa suit that cannot be removed.
- Naughty Stare: Krampus freezes naughty children in place. Practically glare. It won't do shit if you're not on the naughty list.
- Teleport to Workshop: Krampus slowly teleports to his workshop.
- Conjure Unholy bells: These are purely cosmetic and lore-y. It's also a reference to the movie.
- A Bag of Horrors.
  - Activating it in your hand shakes the bag around, stunning anyone inside.
  - You have to drag drop people into it. Naughty children (marked by Santa) will get dropped down 2x as fast.
  - Everyone in Krampus's bag is taken to another z level room.
  - Attack a cursed toy (in Krampus's workshop) to extract someones soul into it, which will instantly teleport them to the station.
  - If someone who's not Krampus opens it, they get dusted, and than their body shortly restored after a few seconds just as a warning.
- Ethereal Jaunt ability.
- Hellish Fire: it operates like dumbfire. projectile. less explosive. expands a lot when hitting a turf. does not expand when hitting a human.
- Krampus takes 2.1x more burn and brute damage than your average human, making lasers and toolboxes more efficient against the demon.

Eventually after a certain period of time, Krampus will be self-defeated if prolonged long enough. Everyone stored in his bag is revived, and placed on the station. Everyone turned into a toy is returned to their body, revived, and put on the station. And then non copyrighted christmas music is played. 

The same thing also happens whenever Krampus's health reaches 0 or lower (aka crit).

 (comparison between Krampus and Santa) krampus: sprites by KMC

(where everyone in his bag is sent)

 (Krampus's Lair with cursed toys)

If Krampus is too overwhelming, we could make him spawn maybe 45 minutes to an hour into the round instead of however long he spawns now.

##### Changelog

:cl:
rscadd: IT'S SUPER'S CHRISTMAS BASH! HAPPY HOLIDAYS EVERYBODY!
rscadd: Enjoy the eggnog at the bar!
rscadd: A shot of eggnog creates a Christmas Miracle!
rscadd: NEW MAP - FULL OF SNOW AND DECORATED CHRISTMAS TREES!
rscadd: Santa can now assign people to the naughty list.
rscadd: The Grinch, and the legion of whatchamaycallits, are here to ruin Christmas! Be on the alert for a green man running around or perhaps an abductor looking fellow.
rscadd: The Christmas Krampus has been awakened. He runs amok on the station. Be careful to double check who's REALLY under that santa costume.
/:cl:

### CODE REVIEW

**Comment by Super3222** [comment]
# An explanation for the map changes.

What I did was this:
I did not replace our current map with Utah's.
I added it to the folder Tgstation - now called Yogstation instead.
I than went into Tgstation.dm (which is where you can control which maps load, etc etc), and re-arranged some pathing so that the snow map spawned instead of the normal one.

We'll definitely have to rearrange this in the future because it's going to get really old, really fast or figure out a way to always go to that map during Christmas.

Speaking of Christmas, it was extended till the 28th. I would've struck for 31st, but that's a little far from... well today. Christmas.

**Comment by Jc1469** [comment]
> map is now white

I'm all for it's-not-christmas-anymore christmas-crap but making the map white >_>

Lavaland is bad enough with the _water_ burning you to death

**Comment by Super3222** [comment]
@Jc1469 I didn't make the map, but I can still replace the majority of the turfs.... I'd be glad to accept not-so-white floors.

File: code/modules/events/holiday/xmas.dm [hunk 1]
```diff
@@ -80,73 +107,593 @@
...
+			if(response == "Yes" && M && M.client && M.stat == DEAD && !chosen)
+				dressup(M)
+
+/datum/round_event/holidayrole/proc/dressup(var/mob/M)
+	return
+
+
+/* 			******************* CHRISTMAS ROLE DEFINES ***************************
+ Santa - Runs around, gives people presents, assigns bad people to the naughty list, hears all.
+ Grinch - Runs around, destroys Christmas, turns good presents / gifsts into SORDs, the worst creation ever.
```
Comment (Jc1469) [inline code/modules/events/holiday/xmas.dm]: > gifsts

probably more typos but this one is one that glowed very brightly as I skimmed through.
but I guess it's a comment :^)

File: code/modules/events/holiday/xmas.dm [hunk 2]
```diff
@@ -80,73 +107,593 @@
...
+	santa_objective.completed = 1 //lets cut our santas some slack.
+	santa_objective.owner = santa.mind
+	santa.mind.objectives += santa_objective
+	santa.mind.AddSpell(new /obj/effect/proc_holder/spell/aoe_turf/conjure/presents)
+	santa.mind.AddSpell(new /obj/effect/proc_holder/spell/targeted/coal)
+	var/obj/effect/proc_holder/spell/targeted/area_teleport/teleport/telespell = new(santa)
+	telespell.clothes_req = 0 //santa robes aren't actually magical.
+	santa.mind.AddSpell(telespell) //does the station have chimneys? WHO KNOWS!
+
+	santa << "<span class='boldannounce'>You are Santa! Your objective is to bring joy to the people on this station. and to dictate whether someone is naughty or nice. You can conjure more presents using a spell, and there are several presents in your bag. Merry Christmas!</span>"
```
Comment (Jc1469) [inline code/modules/events/holiday/xmas.dm]: fullstop after station and in an actual message people see :cry:

File: code/modules/events/holiday/xmas.dm [hunk 3]
```diff
@@ -80,73 +107,593 @@
...
+	name = "Krampus has been summoned (Christmas)"
+	holidayID = CHRISTMAS
+	typepath = /datum/round_event/holidayrole/krampus
+	weight = 100
+	max_occurrences = 1
+	earliest_start = 11000
+
+/datum/round_event/holidayrole/krampus
+	var/mob/living/carbon/human/krampus
+	windowtext1 = "Krampus has been summoned"
```
Comment (Jc1469) [inline code/modules/events/holiday/xmas.dm]: fullstop?

File: code/modules/events/holiday/xmas.dm [hunk 4]
```diff
@@ -80,73 +107,593 @@
...
+	krampus.mind.AddSpell(new /obj/effect/proc_holder/spell/aoe_turf/conjure/unholybells)
+	var/obj/effect/proc_holder/spell/dumbfire/fireball/hellish/H = new(src) // this is too fast.
+	H.charge_max = 600
+	H.cooldown_min = 500
+	krampus.mind.AddSpell(new /obj/effect/proc_holder/spell/dumbfire/fireball/hellish)
+	var/obj/effect/proc_holder/spell/targeted/ethereal_jaunt/eth = new(krampus) // maybe the devils jaunt is better... hmmm
+	eth.clothes_req = 0
+	krampus.mind.AddSpell(eth)
+
+	krampus << "<span class='boldannounce'>You are the Christmas Krampus! Your job is to punish the naughty personell of SS13, and to bring sinners to justice by turning them into your puppets!</span>"
```
Comment (Jc1469) [inline code/modules/events/holiday/xmas.dm]: personnel not personell

File: code/modules/events/holiday/xmas.dm [hunk 5]
```diff
@@ -80,73 +107,593 @@
...
+	krampus << "<span class='notice'>While you are an antagonist, you are not here to kill or attack anyone. Your purpose is only to punish naughty children.</span>"
+
+	// CHRIST-, I MEAN... KRAMPUS DOES NOT LAST FOREVER
+	spawn(10000)
+		activekrampus << "<span class='warning'>You can feel your power crumbling...It is almost time to vanish. You will retreat to your lair for the rest of the shift very soon</span>"
```
Comment (Jc1469) [inline code/modules/events/holiday/xmas.dm]: fullstop pls

File: code/modules/events/holiday/xmas.dm [hunk 6]
```diff
@@ -80,73 +107,593 @@
...
+				sacarea = L
+
+/obj/item/weapon/bagofhorrors/attack_self(mob/user)
+	if(!iskrampus(user))
+		if(user.mind)
+			if(user.mind.assigned_role == "Santa")
+				for(var/mob/M in stored_mobs)
+					user << "<span class='notice'>[M], the [M.job], is sitting inside.</span>"
+				return
+		user << "<span class='warning'>You slowly lookin inside....</span><span clsss='genesisred'>AND A BLAST FLIES \
```
Comment (Jc1469) [inline code/modules/events/holiday/xmas.dm]: lookin?

**Review by Jc1469 — COMMENTED** [review]
overall is cool shit. pls fix gramoor and speel

**Pull request merged**